From: Alexandre Julliard <julliard@winehq.org>
Date: Mon, 21 Jul 2008 12:47:55 +0000 (+0200)
Subject: Avoid exporting common symbols since that's broken on Mac OS X.
X-Git-Tag: wine-1.1.2~158
X-Git-Url: http://source.winehq.org/git/wine.git/?a=commitdiff_plain;h=9c29dbd9873b41f5334a858d8806a256bf00f880

Avoid exporting common symbols since that's broken on Mac OS X.
---

diff --git a/dlls/crtdll/crtdll_main.c b/dlls/crtdll/crtdll_main.c
index 8e5cc18..60c4264 100644
--- a/dlls/crtdll/crtdll_main.c
+++ b/dlls/crtdll/crtdll_main.c
@@ -37,14 +37,14 @@ extern void CDECL __getmainargs( int *argc, char ***argv, char ***envp,
                                  int expand_wildcards, int *new_mode );
 
 /* The following data items are not exported from msvcrt */
-unsigned int CRTDLL__basemajor_dll;
-unsigned int CRTDLL__baseminor_dll;
-unsigned int CRTDLL__baseversion_dll;
-unsigned int CRTDLL__cpumode_dll;
-unsigned int CRTDLL__osmajor_dll;
-unsigned int CRTDLL__osminor_dll;
-unsigned int CRTDLL__osmode_dll;
-unsigned int CRTDLL__osversion_dll;
+unsigned int CRTDLL__basemajor_dll = 0;
+unsigned int CRTDLL__baseminor_dll = 0;
+unsigned int CRTDLL__baseversion_dll = 0;
+unsigned int CRTDLL__cpumode_dll = 0;
+unsigned int CRTDLL__osmajor_dll = 0;
+unsigned int CRTDLL__osminor_dll = 0;
+unsigned int CRTDLL__osmode_dll = 0;
+unsigned int CRTDLL__osversion_dll = 0;
 
 /* dev_t is a short in crtdll but an unsigned int in msvcrt */
 typedef short crtdll_dev_t;
diff --git a/dlls/msvcrt/data.c b/dlls/msvcrt/data.c
index b0f78d5..fd80210 100644
--- a/dlls/msvcrt/data.c
+++ b/dlls/msvcrt/data.c
@@ -29,34 +29,34 @@
 
 WINE_DEFAULT_DEBUG_CHANNEL(msvcrt);
 
-int MSVCRT___argc;
-unsigned int MSVCRT_basemajor;/* FIXME: */
-unsigned int MSVCRT_baseminor;/* FIXME: */
-unsigned int MSVCRT_baseversion; /* FIXME: */
-unsigned int MSVCRT__commode;
-unsigned int MSVCRT__fmode;
-unsigned int MSVCRT_osmajor;/* FIXME: */
-unsigned int MSVCRT_osminor;/* FIXME: */
-unsigned int MSVCRT_osmode;/* FIXME: */
-unsigned int MSVCRT__osver;
-unsigned int MSVCRT_osversion; /* FIXME: */
-unsigned int MSVCRT__winmajor;
-unsigned int MSVCRT__winminor;
-unsigned int MSVCRT__winver;
-unsigned int MSVCRT___setlc_active;
-unsigned int MSVCRT___unguarded_readlc_active;
-double MSVCRT__HUGE;
-char **MSVCRT___argv;
-MSVCRT_wchar_t **MSVCRT___wargv;
-char *MSVCRT__acmdln;
-MSVCRT_wchar_t *MSVCRT__wcmdln;
-char **MSVCRT__environ = 0;
-MSVCRT_wchar_t **_wenviron = 0;
-char **MSVCRT___initenv = 0;
-MSVCRT_wchar_t **MSVCRT___winitenv = 0;
-int MSVCRT_app_type;
-char* MSVCRT__pgmptr = 0;
-WCHAR* MSVCRT__wpgmptr = 0;
+int MSVCRT___argc = 0;
+unsigned int MSVCRT_basemajor = 0;/* FIXME: */
+unsigned int MSVCRT_baseminor = 0;/* FIXME: */
+unsigned int MSVCRT_baseversion = 0; /* FIXME: */
+unsigned int MSVCRT__commode = 0;
+unsigned int MSVCRT__fmode = 0;
+unsigned int MSVCRT_osmajor = 0;/* FIXME: */
+unsigned int MSVCRT_osminor = 0;/* FIXME: */
+unsigned int MSVCRT_osmode = 0;/* FIXME: */
+unsigned int MSVCRT__osver = 0;
+unsigned int MSVCRT_osversion = 0; /* FIXME: */
+unsigned int MSVCRT__winmajor = 0;
+unsigned int MSVCRT__winminor = 0;
+unsigned int MSVCRT__winver = 0;
+unsigned int MSVCRT___setlc_active = 0;
+unsigned int MSVCRT___unguarded_readlc_active = 0;
+double MSVCRT__HUGE = 0;
+char **MSVCRT___argv = NULL;
+MSVCRT_wchar_t **MSVCRT___wargv = NULL;
+char *MSVCRT__acmdln = NULL;
+MSVCRT_wchar_t *MSVCRT__wcmdln = NULL;
+char **MSVCRT__environ = NULL;
+MSVCRT_wchar_t **_wenviron = NULL;
+char **MSVCRT___initenv = NULL;
+MSVCRT_wchar_t **MSVCRT___winitenv = NULL;
+int MSVCRT_app_type = 0;
+char* MSVCRT__pgmptr = NULL;
+WCHAR* MSVCRT__wpgmptr = NULL;
 
 /* Get a snapshot of the current environment
  * and construct the __p__environ array
diff --git a/dlls/msvcrt/file.c b/dlls/msvcrt/file.c
index f85afd3..cd15e40 100644
--- a/dlls/msvcrt/file.c
+++ b/dlls/msvcrt/file.c
@@ -75,7 +75,7 @@ typedef struct {
 
 static ioinfo MSVCRT_fdesc[MSVCRT_MAX_FILES];
 
-MSVCRT_FILE MSVCRT__iob[3];
+MSVCRT_FILE MSVCRT__iob[3] = { { 0 } };
 
 static int MSVCRT_fdstart = 3; /* first unallocated fd */
 static int MSVCRT_fdend = 3; /* highest allocated fd */
diff --git a/dlls/msvcrt/locale.c b/dlls/msvcrt/locale.c
index e679a28..6447436 100644
--- a/dlls/msvcrt/locale.c
+++ b/dlls/msvcrt/locale.c
@@ -43,11 +43,11 @@ WINE_DEFAULT_DEBUG_CHANNEL(msvcrt);
  */
 #define MAX_ELEM_LEN 64 /* Max length of country/language/CP string */
 #define MAX_LOCALE_LENGTH 256
-char MSVCRT_current_lc_all[MAX_LOCALE_LENGTH];
-LCID MSVCRT_current_lc_all_lcid;
-int MSVCRT___lc_codepage;
-int MSVCRT___lc_collate_cp;
-HANDLE MSVCRT___lc_handle[MSVCRT_LC_MAX - MSVCRT_LC_MIN + 1];
+char MSVCRT_current_lc_all[MAX_LOCALE_LENGTH] = { 0 };
+LCID MSVCRT_current_lc_all_lcid = 0;
+int MSVCRT___lc_codepage = 0;
+int MSVCRT___lc_collate_cp = 0;
+HANDLE MSVCRT___lc_handle[MSVCRT_LC_MAX - MSVCRT_LC_MIN + 1] = { 0 };
 
 /* MT */
 #define LOCK_LOCALE   _mlock(_SETLOCALE_LOCK);
diff --git a/dlls/msvcrt/mbcs.c b/dlls/msvcrt/mbcs.c
index 4acc52f..c99d167 100644
--- a/dlls/msvcrt/mbcs.c
+++ b/dlls/msvcrt/mbcs.c
@@ -30,7 +30,7 @@
 
 WINE_DEFAULT_DEBUG_CHANNEL(msvcrt);
 
-unsigned char MSVCRT_mbctype[257];
+unsigned char MSVCRT_mbctype[257] = { 0 };
 static int g_mbcp_is_multibyte = 0;
 
 int MSVCRT___mb_cur_max = 1;
diff --git a/dlls/ntoskrnl.exe/ntoskrnl.c b/dlls/ntoskrnl.exe/ntoskrnl.c
index cda3476..564c55b 100644
--- a/dlls/ntoskrnl.exe/ntoskrnl.c
+++ b/dlls/ntoskrnl.exe/ntoskrnl.c
@@ -40,7 +40,7 @@ WINE_DEFAULT_DEBUG_CHANNEL(ntoskrnl);
 WINE_DECLARE_DEBUG_CHANNEL(relay);
 
 
-KSYSTEM_TIME KeTickCount;
+KSYSTEM_TIME KeTickCount = { 0, 0, 0 };
 
 typedef struct _KSERVICE_TABLE_DESCRIPTOR
 {
@@ -50,7 +50,7 @@ typedef struct _KSERVICE_TABLE_DESCRIPTOR
     PUCHAR Number;
 } KSERVICE_TABLE_DESCRIPTOR, *PKSERVICE_TABLE_DESCRIPTOR;
 
-KSERVICE_TABLE_DESCRIPTOR KeServiceDescriptorTable[4];
+KSERVICE_TABLE_DESCRIPTOR KeServiceDescriptorTable[4] = { { 0 } };
 
 typedef void (WINAPI *PCREATE_PROCESS_NOTIFY_ROUTINE)(HANDLE,HANDLE,BOOLEAN);
 
